r/src/table.cpp

```diff
@@ -117,54 +117,84 @@ std::shared_ptr<arrow::Table> Table__from_dots(SEXP lst, SEXP schema_sxp) {
     return tab;
   }
 
-  R_xlen_t n = XLENGTH(lst);
-  std::vector<std::shared_ptr<arrow::Column>> columns(n);
+  int num_fields;
```

Comment:
I'm curious why the record_batch and table code isn't more shared

Reply:
It could be. record batch only have to handle arrays, where tables have to handle arrays, chunked arrays, columns. but I agree that the structure of the code is similar.

Reply:
I think we should refactor the flatten-for loop into a small function if they're all the same.